src/ngamsUtils/ngamsUtils/ngasXSyncTool.py

```diff
@@ -1281,18 +1301,19 @@ def execute(param_dict):
         streams = int(param_dict[PAR_STREAMS]) + 1
     else:
         streams = int(param_dict[PAR_STREAMS])
-    sync_threads = ngamsThreadGroup.ngamsThreadGroup(param_dict[PAR_SESSION_ID], sync_thread, streams, [param_dict])
-    sync_threads.start()
+    global sync_thread_group
+    sync_thread_group = ngamsThreadGroup.ngamsThreadGroup(param_dict[PAR_SESSION_ID], sync_thread, streams, [param_dict])
+    sync_thread_group.start()
```

Comment:
Instead of making `sync_thread_group` a global variable (because you want to be able to see it from `signal_handler` I presume), I'd suggest you define the signal handler function *within* this function (it can be done). Doing so will give `signal_handler` visibility over the scope of the enclosing function, so there wouldn't be a need for globals anymore.

And here is where you'd want to set the signal handle probably too.

Reply:
OK. I think I understand what you are suggesting. I have applied changes.